tests/sentry/quotas/redis/tests.py

```diff
@@ -81,6 +81,49 @@ def test_is_rate_limited_script():
 class RedisQuotaTest(TestCase):
     quota = fixture(RedisQuota)
 
+    def test_project_abuse_quotas(self):
```

Comment:
Is this test sufficient?

I don't see any relay integration tests with quotas, wonder if it'd just be a pain to write reliable tests for that. Currently I've just been sending 100 events real fast and setting like, error abuse limit to 20 and seeing that only 20 events made it through (I _could_ learn how to query outcomes in clickhouse, I think that's the real way to verify).

Reply:
Also need to test like,

OrganizationOption.objects.set_value(o, "sentry:project-rate-limit", 50)
OrganizationOption.objects.set_value(o, "sentry:account-rate-limit", 100)

And then see if I'm allowed 50% * 100 = 50 events or 20.

Or... should we just remove the legacy RLs?

Reply:
You can use a `relay_integration` E2E test:

https://github.com/getsentry/sentry/blob/d15f54fba007c52a3b1d11dc34cb6a13dad2c4dc/tests/relay_integration/test_integration.py#L14-L18

I'd suggest to set the quota to `0` which avoids any flakyness (even with a quota of `1` you might roll over the window), but it does the job in ensuring a roundtrip.

It's on the heavy side to run but the ideal smoke test for this, and you probably just need a single one. Only thing you couldn't cover with this is the outcomes consumers, which is acceptable. 

--

As for legacy rate limits, I wouldn't worry about those too much. There is already coverage on how multiple rate limits interact with each other, and you shouldn't disable one rate limit because of another. Rate limits are completely additive.